### ISSUE TITLE
Edge case crashes in galleries with GIFs or videos

### ISSUE THREAD

**Issue by 9-5** [9-5/Zennit#3]
Example of gif Gallery crash:
![image](https://github.com/user-attachments/assets/891c8ff4-011d-49d9-b28b-d9edac8d9d5a)
https://www.reddit.com/r/test/comments/1hc4wpg/gallery_test/

```
undefined is not an object (evaluating 'media.s.u.replace')
```

**Comment by 9-5** [comment]
Likely causes the same problem with videos in the gallery

Here's the expected Reddit JSON:

```json
[{"kind": "Listing", "data": {"after": null, "dist": 1, "modhash": "", "geo_filter": "", "children": [{"kind": "t3", "data": {"approved_at_utc": null, "subreddit": "test", "selftext": "", "user_reports": [], "saved": false, "mod_reason_title": null, "gilded": 0, "clicked": false, "is_gallery": true, "title": "Gallery test", "link_flair_richtext": [], "subreddit_name_prefixed": "r/test", "hidden": false, "pwls": 6, "link_flair_css_class": null, "downs": 0, "thumbnail_height": 78, "top_awarded_type": null, "name": "t3_1hc4wpg", "media_metadata": {"wcvss6kcha6e1": {"status": "valid", "e": "AnimatedImage", "m": "image/gif", "p": [{"y": 60, "x": 108, "u": "https://preview.redd.it/wcvss6kcha6e1.gif?width=108&amp;crop=smart&amp;format=png8&amp;s=bcd27398fcc8c15a6980b79172f4b5853a0ee3a5"}, {"y": 121, "x": 216, "u": "https://preview.redd.it/wcvss6kcha6e1.gif?width=216&amp;crop=smart&amp;format=png8&amp;s=5a24141e74b9ee2cf71b3132659b7e7438f3b52f"}, {"y": 179, "x": 320, "u": "https://preview.redd.it/wcvss6kcha6e1.gif?width=320&amp;crop=smart&amp;format=png8&amp;s=9192c75c4fa556e9a035e1b8bc309bf5c84b6d49"}], "s": {"y": 267, "gif": "https://i.redd.it/wcvss6kcha6e1.gif", "mp4": "https://preview.redd.it/wcvss6kcha6e1.gif?format=mp4&amp;s=eb23876f602e8de83b4e71e61b6229e33419178c", "x": 476}, "id": "wcvss6kcha6e1"}, "q46px7kcha6e1": {"status": "valid", "e": "Image", "m": "image/jpg", "p": [{"y": 110, "x": 108, "u": "https://preview.redd.it/q46px7kcha6e1.jpg?width=108&amp;crop=smart&amp;auto=webp&amp;s=13ecbfeb8d9392d169a8407f88b31354fdd41c66"}, {"y": 221, "x": 216, "u": "https://preview.redd.it/q46px7kcha6e1.jpg?width=216&amp;crop=smart&amp;auto=webp&amp;s=dfab1c75337ab7921a51b6a52d13918d6d1d9e24"}, {"y": 328, "x": 320, "u": "https://preview.redd.it/q46px7kcha6e1.jpg?width=320&amp;crop=smart&amp;auto=webp&amp;s=86352ed8efc0268b2be1aee16a867ac34435b665"}, {"y": 657, "x": 640, "u": "https://preview.redd.it/q46px7kcha6e1.jpg?width=640&amp;crop=smart&amp;auto=webp&amp;s=279b8d8610feb58d7be1ebb947f74485e0b58dde"}], "s": {"y": 657, "x": 640, "u": "https://preview.redd.it/q46px7kcha6e1.jpg?width=640&amp;format=pjpg&amp;auto=webp&amp;s=f0523e4585db7d4b432e324fd29751f9d1c08342"}, "id": "q46px7kcha6e1"}}, "hide_score": false, "quarantine": false, "link_flair_text_color": "dark", "upvote_ratio": 1.0, "author_flair_background_color": null, "ups": 1, "domain": "reddit.com", "media_embed": {}, "thumbnail_width": 140, "author_flair_template_id": null, "is_original_content": false, "author_fullname": "t2_334td", "secure_media": null, "is_reddit_media_domain": false, "is_meta": false, "category": null, "secure_media_embed": {}, "gallery_data": {"items": [{"media_id": "wcvss6kcha6e1", "id": 566304862}, {"media_id": "q46px7kcha6e1", "id": 566304863}]}, "link_flair_text": null, "can_mod_post": false, "score": 1, "approved_by": null, "is_created_from_ads_ui": false, "author_premium": false, "thumbnail": "https://a.thumbs.redditmedia.com/n4bj9Qej6_QNEkIaXO4LupTJbnrCa7MvfAVfXqPDNC8.jpg", "edited": false, "author_flair_css_class": null, "author_flair_richtext": [], "gildings": {}, "content_categories": null, "is_self": false, "subreddit_type": "public", "created": 1733953293.0, "link_flair_type": "text", "wls": 6, "removed_by_category": null, "banned_by": null, "author_flair_type": "text", "total_awards_received": 0, "allow_live_comments": false, "selftext_html": null, "likes": null, "suggested_sort": null, "banned_at_utc": null, "url_overridden_by_dest": "https://www.reddit.com/gallery/1hc4wpg", "view_count": null, "archived": false, "no_follow": true, "is_crosspostable": false, "pinned": false, "over_18": false, "all_awardings": [], "awarders": [], "media_only": false, "can_gild": false, "spoiler": false, "locked": false, "author_flair_text": null, "treatment_tags": [], "visited": false, "removed_by": null, "mod_note": null, "distinguished": null, "subreddit_id": "t5_2qh23", "author_is_blocked": false, "mod_reason_by": null, "num_reports": null, "removal_reason": null, "link_flair_background_color": "", "id": "1hc4wpg", "is_robot_indexable": true, "num_duplicates": 0, "report_reasons": null, "author": "edgan", "discussion_type": null, "num_comments": 0, "send_replies": true, "media": null, "contest_mode": false, "author_patreon_flair": false, "author_flair_text_color": null, "permalink": "/r/test/comments/1hc4wpg/gallery_test/", "stickied": false, "url": "https://www.reddit.com/gallery/1hc4wpg", "subreddit_subscribers": 24904, "created_utc": 1733953293.0, "num_crossposts": 0, "mod_reports": [], "is_video": false}}], "before": null}}, {"kind": "Listing", "data": {"after": null, "dist": null, "modhash": "", "geo_filter": "", "children": [], "before": null}}]
```

**Comment by 9-5** [comment]
Issue has been resolved in commit [a9e631b607eb3e732286162bef10fd60a7f64a2d](https://github.com/9-5/Zennit/commit/a9e631b607eb3e732286162bef10fd60a7f64a2d)
